test/dataloader/ecto_test.exs

```diff
@@ -38,13 +38,13 @@ defmodule Dataloader.EctoTest do
 
     User
     |> where(^Enum.to_list(args))
-    |> then(fn user ->
+    |> (fn user ->
       if is_nil(sort_by) or is_nil(sort_order) do
         user
       else
         order_by(user, {^sort_order, ^sort_by})
       end
-    end)
+    end).()
```

Comment:
The current pipeline of this repo is failing because of the Kernel.then/2 function(error: undefined function then/2). 
The existing pipeline has jobs for Elixir 1.1X and OTP 22. 
Whereas The Kernel.then/2 function was added with Elixir.1.12
https://github.com/absinthe-graphql/dataloader/actions/runs/2652520264/jobs/4121762498.

I have replaced the then/2 function with an anonymous function to fix current the pipeline.

@benwilson512